### PR TITLE
Improved ECS attribute and origin translation in awsxrayexporter

### PR DESCRIPTION
**Pull request by willarmiros**
**Description**
Follows up #1360 by translating the relevant ECS attributes from their OTel versions to X-Ray Segment attributes within the `aws.ecs` block. Also added the two new ECS origins that are derived from the ECS Launch Type. I updated the origin resolution logic to prefer the `cloud.infrastructure_service` attribute when available instead of guessing the cloud service type based on other predefined attributes. This PR also introduces the `cloudwatch_logs` field to the AWS translation model, since they are recorded in the ECS resource detector with other services planned. 

I will replace all the strings for resource attribute names with more officially defined constants once https://github.com/open-telemetry/opentelemetry-specification/pull/1112 and https://github.com/open-telemetry/opentelemetry-specification/pull/1099 are merged. 

**Minor changes**
* Updated the origin discovery to return an empty origin if no relevant attributes were discovered, instead of returning the EC2 origin by default
* Included the `cloud.infrastructure_service` attribute in the EC2 resource detector. I'll avoid adding it to non-AWS detectors til it's an official attribute
* Changed ECS resource detector to record Cluster ARN instead of Cluster Name to be consistent with other resources recorded

**Testing:** Added unit tests/updated existing ones. Also tested end-to-end with a sample ECS app publishing to X-Ray.

**Documentation:** Updated resource detector Readme for its minor changes.

### CODE REVIEW

File: exporter/awsxrayexporter/translator/aws.go [hunk 1]
```diff
@@ -61,6 +71,8 @@ func makeAws(attributes map[string]string, resource pdata.Resource) (map[string]
 			switch key {
 			case semconventions.AttributeCloudProvider:
 				cloud = value.StringVal()
+			case "cloud.infrastructure_service":
```
Comment (anuraaga) [inline exporter/awsxrayexporter/translator/aws.go:74]: As these are destined to be replaced by constants anyways, I'd go ahead and have private constants like `AttributeCloudInfrastructureService` defined in this file

File: exporter/awsxrayexporter/translator/aws.go [hunk 2]
```diff
@@ -163,6 +196,14 @@ func makeAws(attributes map[string]string, resource pdata.Resource) (map[string]
 		}
 	}
 
+	// Since we must couple log group ARNs and Log Group Names in the same CWLogs object, we first try to derive the
+	// names from the ARN, then fall back to just recording the names
+	if logGroupArns != (pdata.AnyValueArray{}) && logGroupArns.Len() > 0 {
```
Comment (anuraaga) [inline exporter/awsxrayexporter/translator/aws.go:201]: Do we need this `!=` even though we have a len check?

File: exporter/awsxrayexporter/translator/segment.go [hunk 3]
```diff
@@ -235,6 +237,38 @@ func determineAwsOrigin(resource pdata.Resource) string {
 			return ""
 		}
 	}
+
+	// TODO(willarmiros): Only use infrastructure_service for origin resolution once detectors for all AWS environments are
+	// implemented for robustness
+	is, present := resource.Attributes().Get("cloud.infrastructure_service")
```
Comment (anuraaga) [inline exporter/awsxrayexporter/translator/segment.go:243]: I think you can inline, e.g., `if is, ok := resource...; ok {`

File: processor/resourcedetectionprocessor/internal/aws/ecs/ecs.go [hunk 4]
```diff
@@ -92,10 +91,10 @@ func (d *Detector) Detect(context.Context) (pdata.Resource, error) {
 	// The launch type and log data attributes are only available in TMDE v4
 	switch lt := strings.ToLower(tmdeResp.LaunchType); lt {
 	case "ec2":
-		attr.InsertString("aws.ecs.launchtype", "EC2")
+		attr.InsertString("aws.ecs.launchtype", "ec2")
```
Comment (anuraaga) [inline processor/resourcedetectionprocessor/internal/aws/ecs/ecs.go:94]: If it's possible, it'd generally be better to have separate PRs for this processor and the exporter

**Review by anuraaga — COMMENTED** [review]
Just some small points, thanks

File: exporter/awsxrayexporter/translator/aws.go (hunk 2 above)
Comment (willarmiros) [reply exporter/awsxrayexporter/translator/aws.go:201]: Yeah, I initially didn't have it and it turns out the `Len()` method relies on an underlying pointer `orig` in the `AnyValueArray` struct, so if we don't initialize `logGroupArns` we get a nil pointer exception. I could check for nilness of `orig`, but it seemed like an implementation detail that I shouldn't be referencing directly.

File: exporter/awsxrayexporter/translator/aws.go (hunk 1 above)
Comment (willarmiros) [reply exporter/awsxrayexporter/translator/aws.go:74]: That's fair, will replace

File: processor/resourcedetectionprocessor/internal/aws/ecs/ecs.go (hunk 4 above)
Comment (willarmiros) [reply processor/resourcedetectionprocessor/internal/aws/ecs/ecs.go:94]: Sorry, definitely makes sense. In this case I realized some issues with the detector as I was making the exporter changes, and made them side-by-side. In the future I will definitely take care to separate PRs by module.

**Review by anuraaga — APPROVED** [review]
Thanks!

**Comment by tigrannajaryan** [comment]
@willarmiros please resolve the conflicts.

**Comment by willarmiros** [comment]
@tigrannajaryan done, sorry for the delay

**Comment by willarmiros** [comment]
Looks like https://github.com/open-telemetry/opentelemetry-collector/pull/2039 removed some APIs that broke this PR and #1360 which is already merged. I'll get them fixed so the tests pass.
